src/schema/campaign-notification.js

```diff
@@ -0,0 +1,44 @@
+const { Schema } = require('mongoose');
+const connection = require('../connections/mongoose/instance');
+
+const schema = new Schema({
+  campaignId: {
+    type: Schema.Types.ObjectId,
+    required: true,
+    validate: {
+      async validator(v) {
+        const doc = await connection.model('campaign').findOne({ _id: v }, { _id: 1 });
```

Comment:
Change to `.findById(v, { _id: 1 })`
My bad for not doing that before you Cmd C+V